components/camel-jackson/src/main/java/org/apache/camel/component/jackson/AbstractJacksonDataFormat.java

```diff
@@ -157,7 +158,11 @@ public void marshal(Exchange exchange, Object graph, OutputStream stream) throws
         if (this.schemaResolver != null) {
             schema = this.schemaResolver.resolve(exchange);
         }
-        this.objectMapper.writerWithView(jsonView).with(schema).writeValue(stream, graph);
+        // using OutputStreamWriter because of Jackson not handling 4-byte characters properly with only a stream
+        // https://github.com/FasterXML/jackson-core/issues/223
+        OutputStreamWriter writer = new OutputStreamWriter(stream);
+        this.objectMapper.writerWithView(jsonView).with(schema).writeValue(writer, graph);
+        writer.close();
```

Comment:
Use try-with-resources here. It's cleaner.

Reply:
The time I spent doing fixes instead of developing finally starts to show up :) Thanks for the tip!